helix-core/src/main/java/org/apache/helix/manager/zk/ZkBaseDataAccessor.java

```diff
@@ -326,6 +350,30 @@ public T get(String path, Stat stat, int options) {
     return data;
   }
 
+  /**
+   * Sync get method with custom serializer support
+   */
+  public Object get(String path, Stat stat, int options, ZkSerializer serializer) {
+    byte[] content = null;
+    try {
+      content = _nonZNRecordClient.readData(path, stat);
+      return serializer.deserialize(content);
+    } catch (ZkNoNodeException ex) {
```

Comment:
Will ZkMarshallingError (failed to deserialize) be caught in the users of get()?

Reply:
Reminder to remove catch clause for ZkMarshallingError.